code/game/objects/structures/tables_racks.dm

```diff
@@ -154,6 +237,56 @@
 				user.stop_pulling()
 	return ..()
 
+/obj/structure/table/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(!istype(user) || !user.can_interact_with(src))
+		return FALSE
+
+	if(!can_flip)
+		return
+
+	if(!is_flipped)
+		user.balloon_alert_to_viewers("flipping table...")
+		if(!do_after(user, max_integrity * 0.25))
+			return
+
+		//change icons
+		layer = LOW_ITEM_LAYER
+		var/new_dir = get_dir(user, src)
+		if(new_dir == NORTHEAST || new_dir == SOUTHEAST) // Dirs need to be part of the 4 main cardinal directions so proc/CanAllowThrough isn't fucky wucky
+			new_dir = EAST
+		if(new_dir == NORTHWEST || new_dir == SOUTHWEST)
+			new_dir = WEST
+		dir = new_dir
+		if(new_dir == SOUTH)
+			layer = ABOVE_MOB_LAYER
+
+		is_flipped = TRUE
+		flip_table()
+
+	else
+		user.balloon_alert_to_viewers("flipping table upright...")
+		if(do_after(user, max_integrity * 0.25, src))
+			is_flipped = FALSE
+			unflip_table()
+			//icons
+			playsound('sound/items/trayhit/trayhit2.ogg', 100)
+			return TRUE
+
+/obj/structure/table/proc/is_able_to_throw(obj/structure/table, atom/movable/movable_entity)
+	if (movable_entity == table) //Thing is not the table
+		return FALSE
+	if (movable_entity.anchored) //Thing isn't anchored
+		return FALSE
+	if (movable_entity.invisibility > SEE_INVISIBLE_LIVING) //Thing isnt incorporeal ie. ghosts/revenants
+		return FALSE
+	if(!ismob(movable_entity) && !isobj(movable_entity)) //Thing isn't an obj or mob
+		return FALSE
+	if(movable_entity.throwing || (movable_entity.movement_type & (FLOATING|FLYING))) //Thing isn't flying/floating
+		return FALSE
+
```

Comment:
Check for TRAIT_MOB_ELEVATED on mobs please, so it works with TRAIT_IGNORE_ELEVATION (like Melbert's antigravity PR)

Reply:
I was also going to suggest this until I noticed that you did; this means that you can flip tables with xeno larva underneath to reveal them which is very soulful.